CFE-2887: Canonify class names set by module protocol

nickanderson opened the pull request:
Superceeds PR #3273 (this one comes with changelog keyword and acceptance test)

cf-bottom commented:
Thanks for submitting a PR! Maybe @vpodzime can review this?

cf-bottom commented:
Sure, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=1191)](https://ci.cfengine.com//job/pr-pipeline/1191/)

https://ci.cfengine.com/job/pr-pipeline/1191/

nickanderson commented:
Failing the new acceptance test in travis: https://travis-ci.org/cfengine/core/jobs/431589969#L3090

nickanderson commented:
@tzz The patch is failing the test I wrote. Feel free to review the test if you like.

vpodzime commented on libpromises/evalfunction.c:
```diff
@@ -7876,6 +7876,7 @@ void ModuleProtocol(EvalContext *ctx, char *command, const char *line, int print
         if (CheckID(content))
         {
             Buffer *tagbuf = StringSetToBuffer(tags, ',');
+            CanonifyNameInPlace(content);
```
I think we need to add some verbose logging here as well because otherwise the above log message may be confusing/misleading if canonification (?) happens afterwards.

nickanderson replied:
OK.

nickanderson replied:
Logging added

tzz commented:
@nickanderson 
> The patch is failing the test I wrote. Feel free to review the test if you like.

The test looks good to me.

nickanderson commented:
On closer inspection, I found that the canonification needed to be done inside CheckID. I changed the existing error log to WARNING and added VERBOSE logs about automatic canonification.

nickanderson commented:
@cf-bottom jenkins, please

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=1502)](https://ci.cfengine.com//job/pr-pipeline/1502/)

https://ci.cfengine.com/job/pr-pipeline/1502/

olehermanse commented on libpromises/evalfunction.c:
```diff
@@ -7885,6 +7885,15 @@ void ModuleProtocol(EvalContext *ctx, char *command, const char *line, int print
+          Log(LOG_LEVEL_VERBOSE, "Automatically canonified to '%s'", content);
+          Buffer *tagbuf = StringSetToBuffer(tags, ',');
+          EvalContextClassPutSoft(ctx, content, CONTEXT_SCOPE_NAMESPACE, BufferData(tagbuf));
+          BufferDestroy(tagbuf);
```
Indentation is wrong. We use 4 spaces in C code.
